client/js/lounge.js

```diff
@@ -472,16 +472,19 @@ $(function() {
 		// Remove the date-change marker we put at the top, because it may
 		// not actually be a date change now
 		var children = $(chan).children();
-		if (children.eq(0).hasClass("date-marker")) { // Check top most child
+		var dateMarkerHeight = 0;
+		if (children.eq(0).hasClass("date-marker-container")) { // Check top most child
+			dateMarkerHeight = children.eq(0).height();
 			children.eq(0).remove();
-		} else if (children.eq(0).hasClass("unread-marker") && children.eq(1).hasClass("date-marker")) {
-			// Otherwise the date-marker would get 'stuck' because of the new-message marker
+		} else if (children.eq(0).hasClass("unread-marker") && children.eq(1).hasClass("date-marker-container")) {
+			// Check for unread-marker too - oherwise the date-marker would get 'stuck' because of the new-message marker
+			dateMarkerHeight = children.eq(1).height();
 			children.eq(1).remove();
 		}
 
 		// get the scrollable wrapper around messages
 		var scrollable = chan.closest(".chat");
-		var heightOld = chan.height();
+		var heightOld = chan.height() + dateMarkerHeight; // Add the date marker's height because it was already removed
```

Comment:
I guess I was trying to not assume anything, but it is a guarantee that there will be one at the top, so thanks. Moved calculating the height to before I remove it.